### PR TITLE
Remove runtime classpath scanning

### LINKED ISSUE #784: Crash opening sample grip file in 2017 vision example zip file
Installed latest grip and tried to open the example grip pipeline in the 2017 vision example zip file.

### LINKED ISSUE #834: Error opening 2017 Vision Processing files
GRIP does not open the 2017 Vision Example files from [TeamForge](https://usfirst.collab.net/sf/frs/do/viewRelease/projects.wpilib/frs.sample_programs.2017_c_java_vision_sample?_message=1483834990405)
It then freezes up (Not Responding).

This error is exactly the same as #784. 

Error + stacktrace:

`Thread[Project Open Thread,5,main] threw an exception`
## Stack Trace:
```java
com.thoughtworks.xstream.mapper.CannotResolveClassException: grip:Pipeline
	at com.thoughtworks.xstream.mapper.DefaultMapper.realClass(DefaultMapper.java:81)
	at com.thoughtworks.xstream.mapper.MapperWrapper.realClass(MapperWrapper.java:30)
	at com.thoughtworks.xstream.mapper.DynamicProxyMapper.realClass(DynamicProxyMapper.java:55)
	at com.thoughtworks.xstream.mapper.MapperWrapper.realClass(MapperWrapper.java:30)
	at com.thoughtworks.xstream.mapper.PackageAliasingMapper.realClass(PackageAliasingMapper.java:88)
	at com.thoughtworks.xstream.mapper.MapperWrapper.realClass(MapperWrapper.java:30)
	at com.thoughtworks.xstream.mapper.ClassAliasingMapper.realClass(ClassAliasingMapper.java:79)
	at com.thoughtworks.xstream.mapper.MapperWrapper.realClass(MapperWrapper.java:30)
	at com.thoughtworks.xstream.mapper.MapperWrapper.realClass(MapperWrapper.java:30)
	at com.thoughtworks.xstream.mapper.MapperWrapper.realClass(MapperWrapper.java:30)
	at com.thoughtworks.xstream.mapper.MapperWrapper.realClass(MapperWrapper.java:30)
	at com.thoughtworks.xstream.mapper.MapperWrapper.realClass(MapperWrapper.java:30)
	at com.thoughtworks.xstream.mapper.MapperWrapper.realClass(MapperWrapper.java:30)
	at com.thoughtworks.xstream.mapper.ArrayMapper.realClass(ArrayMapper.java:74)
	at com.thoughtworks.xstream.mapper.MapperWrapper.realClass(MapperWrapper.java:30)
	at com.thoughtworks.xstream.mapper.MapperWrapper.realClass(MapperWrapper.java:30)
	at com.thoughtworks.xstream.mapper.MapperWrapper.realClass(MapperWrapper.java:30)
	at com.thoughtworks.xstream.mapper.MapperWrapper.realClass(MapperWrapper.java:30)
	at com.thoughtworks.xstream.mapper.MapperWrapper.realClass(MapperWrapper.java:30)
	at com.thoughtworks.xstream.mapper.MapperWrapper.realClass(MapperWrapper.java:30)
	at com.thoughtworks.xstream.mapper.MapperWrapper.realClass(MapperWrapper.java:30)
	at com.thoughtworks.xstream.mapper.SecurityMapper.realClass(SecurityMapper.java:71)
	at com.thoughtworks.xstream.mapper.MapperWrapper.realClass(MapperWrapper.java:30)
	at com.thoughtworks.xstream.mapper.MapperWrapper.realClass(MapperWrapper.java:30)
	at com.thoughtworks.xstream.mapper.CachingMapper.realClass(CachingMapper.java:47)
	at com.thoughtworks.xstream.core.util.HierarchicalStreams.readClassType(HierarchicalStreams.java:29)
	at com.thoughtworks.xstream.core.TreeUnmarshaller.start(TreeUnmarshaller.java:133)
	at com.thoughtworks.xstream.core.AbstractTreeMarshallingStrategy.unmarshal(AbstractTreeMarshallingStrategy.java:32)
	at com.thoughtworks.xstream.XStream.unmarshal(XStream.java:1230)
	at com.thoughtworks.xstream.XStream.unmarshal(XStream.java:1214)
	at com.thoughtworks.xstream.XStream.fromXML(XStream.java:1085)
	at edu.wpi.grip.core.serialization.Project.open(Project.java:122)
	at edu.wpi.grip.core.serialization.Project.open(Project.java:101)
	at edu.wpi.grip.ui.MainWindowController.lambda$openProject$5(MainWindowController.java:197)
	at java.lang.Thread.run(Thread.java:745)

```
## System Info:

Property Name | Property
 ----- | -----
GRIP Version | v1.5.2
java.version | 1.8.0_101
javafx.version | 8.0.101
os.name | Windows 10
os.version | 10.0
os.arch | amd64

### PR DESCRIPTION
**Pull request by SamCarlberg**
Operations, custom XStream types,and publishable classes are stored in registry files in the META-INF directory of the core project JAR. These files are generated by an annotation processor - when creating a new operation, publishable class, or save-file content, simply add the appropriate annotation to the class (`@Description` , `@PublishableObject` , and `@XStreamAlias`, respectively)

Fixes #784 
Fixes  #834

### CODE REVIEW

File: core/src/main/resources/META-INF/operations [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+edu.wpi.grip.core.operations.composite.RGBThresholdOperation
+edu.wpi.grip.core.operations.composite.FilterContoursOperation
+edu.wpi.grip.core.operations.composite.SaveImageOperation
+edu.wpi.grip.core.operations.composite.ConvexHullsOperation
+edu.wpi.grip.core.operations.composite.FilterLinesOperation
+edu.wpi.grip.core.operations.composite.ThresholdMoving
+edu.wpi.grip.core.operations.composite.PublishVideoOperation
+edu.wpi.grip.core.operations.composite.WatershedOperation
+edu.wpi.grip.core.operations.composite.FindLinesOperation
+edu.wpi.grip.core.operations.composite.DistanceTransformOperation
```
Comment (JLLeitschuh) [inline core/src/main/resources/META-INF/operations]: Can we get Gradle to build this automatically with some sort of annotation processor?

**Comment by JLLeitschuh** [comment]
The other thing to try is just updating Guava. There have been fixes to `ClassPath` in the newer releases that may fix the problems people were having.

File: core/src/main/resources/META-INF/operations (hunk 1 above)
Comment (SamCarlberg) [reply core/src/main/resources/META-INF/operations]: We'd have to write our own annotation processor, and do something for publishables (reports etc) . That'd probably mean creating a new project for the annotations and annotation processor.

**Comment by SamCarlberg** [comment]
Given how difficult it is to reproduce the problem, I don't think that attempting to use a newer version of Guava will be enough to say we've fixed the problem. Especially since `ClassPath` is still marked as beta in the newest version of Guava (27.0.1)

File: annotation/annotation.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+}
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    api(group = "com.google.guava", name = "guava", version = "20.0")
+    api(group = "com.google.auto.service", name = "auto-service", version = "+")
+    annotationProcessor(group = "com.google.auto.service", name = "auto-service", version = "+")
```
Comment (JLLeitschuh) [inline annotation/annotation.gradle.kts]: Do we really want wildcards on these?

**Comment by SamCarlberg** [comment]
@JLLeitschuh Any further comments before I merge?

**Comment by JLLeitschuh** [comment]
Nope, please resolve merge conflicts.

**Pull request merged**